core/src/test/java/org/elasticsearch/search/geo/GeoShapeIntegrationIT.java

```diff
@@ -479,6 +476,40 @@ public void testOrientationPersistence() throws Exception {
         assertThat(orientation, equalTo(ShapeBuilder.Orientation.CCW));
     }
 
+    @Test
+    public void testPointsOnly() throws Exception {
+        String mapping = XContentFactory.jsonBuilder().startObject().startObject("type1")
+                .startObject("properties").startObject("location")
+                .field("type", "geo_shape")
+                .field("tree", randomBoolean() ? "quadtree" : "geohash")
+                .field("tree_levels", "6")
+                .field("distance_error_pct", "0.01")
+                .field("points_only", true)
+                .endObject().endObject()
+                .endObject().endObject().string();
+
+        assertAcked(prepareCreate("geo_points_only").addMapping("type1", mapping));
+        ensureGreen();
+
+        // test exception handling by attempting to insert a random polygon
+        ShapeBuilder shape = RandomShapeGenerator.createShape(random());
```

Comment:
Just so I understand, the random shape could or could not be a ploygon, so if it is a polygon we want to make sure an exception is thrown (and then finish the test), otherwise we want to use it in a query. Correct?